PollBuddy-Server/backend/routes/users.test.js

```diff
@@ -0,0 +1,545 @@
+require("dotenv").config();
+const express = require("express");
+const supertest = require("supertest");
+const bcrypt = require("bcrypt");
+const mongo = require("mongodb");
+const MongoClient = mongo.MongoClient;
+
+var { userSchema } = require("../models/User.js");
+var { createModel } = require("../modules/utils.js");
+var mongoConnection = require("../modules/mongoConnection.js");
+var usersRouter = require("./users");
+const { createTestScheduler } = require("@jest/core");
+const { ExpectationFailed } = require("http-errors");
+
+let mockApp = express();
+let session = {};
+let testUser = {
+  UserName: "test.account",
+  Email: "test@account.com",
+  Password: "K9g95p$?E@t3A$#4",
+  PasswordHash: "$2a$12$8Guj3IMNNVWk/GM4q0xeleExT3QBdPe5dWpSRYvk2elRkkWPMlOPG",
+  FirstName: "test",
+  LastName: "account"
+};
+
+let testUser2 = {
+  UserName: "test.account.2",
+  Email: "test@account2.com",
+  Password: "fbdbxsDzPBo6m68$",
+  PasswordHash: "$2a$12$BQKM4Ml3ag38KacRJRrZhO9FRt5yg4hbp3pg3zF.2ZZPQkn5QIAV2",
+  FirstName: "test.2",
+  LastName: "account.2"
+};
+
+mockApp.use(express.json());
+mockApp.use(express.urlencoded({ extended: false }));
+mockApp.use(function (req, res, next) {
+  req.session = session;
+  next();
+});
+mockApp.use("/api/users", usersRouter); 
+
+let app = supertest(mockApp);
+
+beforeAll(async () => {
+  process.env.DB_URL = global.__MONGO_URI__;
+  process.env.DB_NAME = global.__MONGO_DB_NAME__;
+  process.env.JEST = "true";
+
+  const client = await MongoClient.connect(process.env.DB_URL);
+  const db = client.db(process.env.DB_NAME);
+  mongoConnection.setClient(client);
+  mongoConnection.setDB(db);
+});
+
+afterAll(() => {
+  setTimeout(() => {
+    mongoConnection.getClient().close();
+  }, 1000);
+});
+
+afterEach(async () => {
+  session = {};
+
+  let collections = await mongoConnection.getDB().listCollections().toArray();
+  collections.forEach(async (collection) => {
+    await mongoConnection.getDB().collection(collection.name).deleteMany({});
+  });
+});
+
+let createUser = async function(update) {
+  let userData = {
+    UserName: testUser.UserName,
+    Email: testUser.Email,
+    Password: testUser.PasswordHash,
+    FirstName: testUser.FirstName,
+    LastName: testUser.LastName
+  };
+
+  if (update) {
+    for (let [key, value] of Object.entries(update)) {
+      userData[key] = value;
+    }
+  }
+  
+  let user = createModel(userData, userSchema);
+  let res = await mongoConnection.getDB().collection("users").insertOne(user);
+  return res;
+};
+
+describe("/api/users/login", () => {
+
+  it("GET: fail", async () => {
+    await app.get("/api/users/login")
+      .expect(405)
+      .then((response) => {
+        expect(response.body.result).toBe("failure");
```

Comment:
Check for error message
Create issue to create globally defined / standardized error messages